go/vt/vtgate/evalengine/compiler_asm.go

```diff
@@ -2345,6 +2345,99 @@ func (asm *assembler) Fn_BIT_LENGTH() {
 	}, "FN BIT_LENGTH VARCHAR(SP-1)")
 }
 
+func (asm *assembler) Fn_FIELD(args int, containsOnlyString, containsOnlyInt64 bool) {
+	asm.adjustStack(-args + 1)
+	asm.emit(func(env *ExpressionEnv) int {
+		if env.vm.stack[env.vm.sp-args] == nil {
+			env.vm.stack[env.vm.sp-args] = env.vm.arena.newEvalInt64(0)
+			env.vm.sp -= args - 1
+			return 1
+		}
+
+		if containsOnlyInt64 {
```

Comment:
The idea behind the compiler is that we can emit efficient code once types are known. It looks like these are compile time known values, so we should split this up into separate `Fn_FIELD` functions. Like maybe `Fn_FIELD_i`, `Fn_FIELD_s` etc.